src/buildPipeline.ts

```diff
@@ -40,134 +37,162 @@ export class BuildPipeline implements vscode.Disposable {
     /**
      * Init the build environment
      */
-    async initializeBuild(completeBuild = false) {
-        this.runner.completeBuild = completeBuild
+    async initializeBuild() {
+        this.runner.ensureIdle()
 
         if (this.workspaceState.getInitialized()) {
             console.log('Skipped build initialization. Already initialized.')
             return
         }
 
         await this.manifestManager.doWithActiveManifest(async (activeManifest) => {
-            await this.outputTerminal.show(true)
-            await this.runner.setCommands([activeManifest.initBuild()], TaskMode.buildInit)
+            await this.runner.execute([activeManifest.initBuild()], TaskMode.buildInit)
+            await loadIntegrations(activeManifest)
         })
+
+        await this.workspaceState.setInitialized(true)
```

Comment:
Shouldn't this be done inside the doWithActiveManifest?

Reply:
would it make sense to maybe move the doWithActiveManifest outside the method, then let i take a manifest param

Reply:
That would be better yes